docs/development.md

```diff
@@ -0,0 +1,420 @@
++
+ 	// connections, timeouts:
+ 	idleConnsPerHostUsage           = "maximum idle connections per backend host"
+ 	closeIdleConnsPeriodUsage       = "period of closing all idle connections in seconds or as a duration string. Not closing when less than 0"
+@@ -243,13 +246,14 @@ var (
+ 	oauth2TokeninfoURL              string
+ 	oauth2TokeninfoTimeout          time.Duration
+ 	oauth2TokenintrospectionTimeout time.Duration
++	webhookTimeout                  time.Duration
+
+ 	// connections, timeouts:
+ 	idleConnsPerHost           int
+@@ -351,13 +355,14 @@ func init() {
+ 	flag.DurationVar(&oauth2TokeninfoTimeout, "oauth2-tokeninfo-timeout", defaultOAuthTokeninfoTimeout, oauth2TokeninfoTimeoutUsage)
+ 	flag.DurationVar(&oauth2TokenintrospectionTimeout, "oauth2-tokenintrospect-timeout", defaultOAuthTokenintrospectionTimeout, oauth2TokenintrospectionTimeoutUsage)
++	flag.DurationVar(&webhookTimeout, "webhook-timeout", defaultWebhookTimeout, webhookTimeoutUsage)
+
+ 	// connections, timeouts:
+ 	flag.IntVar(&idleConnsPerHost, "idle-conns-num", proxy.DefaultIdleConnsPerHost, idleConnsPerHostUsage)
+@@ -536,13 +541,14 @@ func main() {
+ 		OAuthTokeninfoURL:              oauth2TokeninfoURL,
+ 		OAuthTokeninfoTimeout:          oauth2TokeninfoTimeout,
+ 		OAuthTokenintrospectionTimeout: oauth2TokenintrospectionTimeout,
++		WebhookTimeout:                 webhookTimeout,
+
+ 		// connections, timeouts:
+ 		IdleConnectionsPerHost:     idleConnsPerHost,
+
+diff --git a/skipper.go b/skipper.go
+index 10d5769..da46fe0 100644
+--- a/skipper.go
++++ b/skipper.go
+@@ -443,6 +443,9 @@ type Options struct {
+ 	// OAuthTokenintrospectionTimeout sets timeout duration while calling oauth tokenintrospection service
+ 	OAuthTokenintrospectionTimeout time.Duration
+
++	// WebhookTimeout sets timeout duration while calling a custom webhook auth service
++	WebhookTimeout time.Duration
++
+ 	// MaxAuditBody sets the maximum read size of the body read by the audit log filter
+ 	MaxAuditBody int
+ }
+@@ -677,7 +680,8 @@ func Run(o Options) error {
+ 		auth.NewOAuthTokenintrospectionAnyClaims(o.OAuthTokenintrospectionTimeout),
+ 		auth.NewOAuthTokenintrospectionAllClaims(o.OAuthTokenintrospectionTimeout),
+ 		auth.NewOAuthTokenintrospectionAnyKV(o.OAuthTokenintrospectionTimeout),
+-		auth.NewOAuthTokenintrospectionAllKV(o.OAuthTokenintrospectionTimeout))
++		auth.NewOAuthTokenintrospectionAllKV(o.OAuthTokenintrospectionTimeout),
++		auth.NewWebhook(o.WebhookTimeout))
+
+ 	// create a filter registry with the available filter specs registered,
+ 	// and register the custom filters
+```
+
+### User documentation
+
+Documentation for users should be done in `docs/`.
+
+```
+diff --git a/docs/filters.md b/docs/filters.md
+index d3bb872..a877062 100644
+--- a/docs/filters.md
++++ b/docs/filters.md
+@@ -382,6 +382,24 @@ basicAuth("/path/to/htpasswd")
+ basicAuth("/path/to/htpasswd", "My Website")
+ ```
```

Comment:
not sure how the code blocks will look in the final output. Have you checked what will be generated? We have example Markdown code blocks in the diff which is already a Markdown code block. Do have any means for escaping?

Reply:
it does not break the markdown